pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -150,19 +151,23 @@ pub async fn handle_walreceiver_connection(
 
                 waldecoder.feed_bytes(data);
 
-                while let Some((lsn, recdata)) = waldecoder.poll_decode()? {
-                    let _enter = info_span!("processing record", lsn = %lsn).entered();
+                {
+                    let mut decoded = DecodedWALRecord::default();
+                    let mut modification = timeline.begin_modification();
+                    while let Some((lsn, recdata)) = waldecoder.poll_decode()? {
+                        // let _enter = info_span!("processing record", lsn = %lsn).entered();
```

Comment:
I see this as a free performance improvement. Can revert the change in this PR if needed

Reply:
The span provides very useful context to any errors or warnings that happen while processing the record. It would be a pity to just outright remove it. However, maybe we can compensate by adding the information another way. I think we could do something like this in the ingest_record call below:
```
    walingest.ingest_record(recdata, lsn, &mut modification, &mut decoded)
  .with_context("could not ingest record at {lsn}")
```

Reply:
Yeah, that would be better.